dev/com.ibm.ws.kernel.service/test/com/ibm/wsspi/kernel/service/utils/MetaTypeUtilsTest.java

```diff
@@ -384,4 +401,289 @@ public void testLocale() throws Exception {
 
         assertEquals(5, (long) MetatypeUtils.evaluateDuration("5\u6beb\u79d2", TimeUnit.MILLISECONDS)); // zh_TW
     }
+
+    @Test
+    public void testScheduleGeneral() throws Exception {
+        SerializableSchedule testSchedule;
+        SerializableSchedule expectedSchedule;
+
+        TreeSet<DayOfWeek> weekdays = new TreeSet<DayOfWeek>();
+        TreeSet<DayOfWeek> alldays = new TreeSet<DayOfWeek>();
+        TreeSet<DayOfWeek> wednesday = new TreeSet<DayOfWeek>();
+
+        Collections.addAll(weekdays, DayOfWeek.MONDAY, DayOfWeek.TUESDAY, DayOfWeek.WEDNESDAY, DayOfWeek.THURSDAY, DayOfWeek.FRIDAY);
+        Collections.addAll(alldays, DayOfWeek.MONDAY, DayOfWeek.TUESDAY, DayOfWeek.WEDNESDAY, DayOfWeek.THURSDAY, DayOfWeek.FRIDAY, DayOfWeek.SATURDAY, DayOfWeek.SUNDAY);
+        Collections.addAll(wednesday, DayOfWeek.WEDNESDAY);
+
+        //Test startup
+        testSchedule = MetatypeUtils.evaluateSchedule("startup")[0];
+        assertTrue(testSchedule.isStartup());
+        assertNull(testSchedule.getValidDays());
+        assertNull(testSchedule.getStartTime());
+        assertNull(testSchedule.getEndTime());
+        assertNull(testSchedule.getTimezone());
+
+        //Test components
+        testSchedule = MetatypeUtils.evaluateSchedule("MON-FRI 8:00-17:00 America/Chicago")[0];
+        assertEquals(weekdays, testSchedule.getValidDays());
+        assertEquals(LocalTime.of(8, 0), testSchedule.getStartTime());
+        assertEquals(LocalTime.of(17, 0), testSchedule.getEndTime());
+        assertEquals(ZoneId.of("America/Chicago"), testSchedule.getTimezone());
+
+        //Test equals
+        expectedSchedule = new SerializableSchedule(DayOfWeek.MONDAY, DayOfWeek.FRIDAY, LocalTime.of(8, 0), LocalTime.of(17, 0), ZoneId.of("America/Chicago"));
+        assertEquals(expectedSchedule, testSchedule);
+
+        //Test Days of week
+        testSchedule = MetatypeUtils.evaluateSchedule("MON-SUN 8")[0];
+        assertEquals(alldays, testSchedule.getValidDays());
+
+        testSchedule = MetatypeUtils.evaluateSchedule("WED-TUES 8")[0];
+        assertEquals(alldays, testSchedule.getValidDays());
+
+        testSchedule = MetatypeUtils.evaluateSchedule("WED-WED 8")[0];
+        assertEquals(wednesday, testSchedule.getValidDays());
```

Comment:
This one is confusing and I wonder if we should reject it.  It isn't clear to me if this should mean all days in the week or only Wednesday.